SharedSource/Core/Utilities/EqualityUtility.cs

```diff
@@ -179,7 +179,7 @@ public static int GetXorHashCode (IEnumerable objects)
     /// Similar to <see cref="Equals{T}"/>, but without any boxing (better performance). 
     /// Equatable objects implement the <see cref="IEquatable{T}"/> interface. 
     /// </remarks>
-    public static bool EqualsEquatable<T> ([AllowNull] T a, [AllowNull] T b)
+    public static bool EqualsEquatable<T> (T? a, T? b)
```

Comment:
revert please. AllowNullAttribute is part of nullable refernece type semantics

Reply:
I really dont like this.

Reply:
I think there was more of a reason to change this than simple preference, I think some builds failed? Doesnt happen anymore though, so its alright.

Reply:
I'm not sure whether I couldn't find a case where it matters or that it has been fixed in the meantime. This is with `T?`:

![image](https://github.com/user-attachments/assets/
caf743d8-1e01-456a-9b63-53992607ea5d)

Reply:
> I'm not sure whether I couldn't find a case where it matters or that it has been fixed in the meantime. 

Hmm... I think you're correct about this being a c# version thing: https://endjin.com/blog/2022/02/csharp-10-generics-nullable-references-improvements-allownull

For tracking purposes, I think a separate task and PR would be better. While this is a very small change on the surface, when reading the semantics, you only understand them with checking up on the change in the c# compiler and a deep dive into why it is okay. Please extract into separate PR.